module/authentication/src/Application/Middleware/AuthenticationMiddleware.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Authentication\Application\Middleware;
+
+use Ergonode\Account\Domain\Entity\User;
+use Ergonode\Account\Domain\Repository\UserRepositoryInterface;
+use Ergonode\Authentication\Application\Stamp\UserStamp;
+use Ergonode\Authentication\Application\Token\UserToken;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Middleware\MiddlewareInterface;
+use Symfony\Component\Messenger\Middleware\StackInterface;
+use Symfony\Component\Messenger\Stamp\ReceivedStamp;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Webmozart\Assert\Assert;
+
+class AuthenticationMiddleware implements MiddlewareInterface
+{
+
+    private TokenStorageInterface $tokenStorage;
+
+    private UserRepositoryInterface $userRepository;
+
+    public function __construct(
+        TokenStorageInterface $tokenStorage,
+        UserRepositoryInterface $userRepository
+    ) {
+        $this->tokenStorage = $tokenStorage;
+        $this->userRepository = $userRepository;
+    }
+
+
+    public function handle(Envelope $envelope, StackInterface $stack): Envelope
+    {
+
+        if ($envelope->last(ReceivedStamp::class)) {
+            $this->tokenStorage->setToken(null);
+            /** @var UserStamp $stamp */
+            $stamp = $envelope->last(UserStamp::class);
+
+            if ($stamp) {
+                $user = $this->userRepository->load($stamp->getUserId());
+                Assert::isInstanceOf(
```

Comment:
I don't think it should throw an exception. I'd create a log error message but not break the entire execution of the message - this should be decided on the individual handling level IMO